### ISSUE TITLE
Can we indicate to objects if they are being 'dummy' constructed?

### ISSUE THREAD

**Issue by pixsperdavid** [Cycling74/min-api#48]
In the case of objects which immediately attempt to make a connection with hardware, being constructed and then immediately destroyed could cause issues. Could we provide some flag that can be tested in the object constructor to determine if this is a 'dummy' construction (just for the purposes of wrapping the object)?

**Comment by tap** [comment]
The above method (in `dict.joiner`) does not work in *nobox* objects, so it is not recommended.

Since we know that objects are always dummy constructed once, we can instead test against a static flag. This code from `link.clocksource`:

```
static bool s_link_clocksource_dummy_construction { true };
```

and later in the ctor:
```
        if (s_link_clocksource_dummy_construction) {
            s_link_clocksource_dummy_construction = false;
            return;
        }
```

**Comment by tap** [comment]
even better. now just call `dummy()` and if it returns true then you are being dummy constructed.

**Comment by impsnldavid** [comment]
Found a possible issue with this. I also need to use `dummy()` in the destructor for my nobox object for lxmax. Because the object is instantiated in the `maxclass_setup` message function, the non-dummy object runs it's constructor and sets the static flag before the dummy object runs its destructor.

Appreciate this is quite a specific case, but perhaps it would be better to flag a specific instance of an object as a dummy at construction time and test against that rather than the static flag?

**Comment by tap** [comment]
Good idea on making `dummy()` function not just for the constructor but for the whole class lifetime.

I think I should be able to do it with the static but setting the flag at destructor time rather than at construction.
